### PR TITLE
Re-sync with the Dart Analysis Server completion protocol

### PR DESCRIPTION
**Pull request by jwren**
https://dart-review.googlesource.com/c/sdk/+/231040 - Add CompletionSuggestion.libraryUri, keep isNotImported.

@alexander-doroshko 
@scheglov 
@bwilkerson

### CODE REVIEW

File: Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java [hunk 1]
```diff
@@ -228,9 +228,9 @@ private static void addToCompletionList(@NotNull Project project,
       if (!addedCompletions.add(suggestion)) continue;
 
       Element element = suggestion.getElement();
-      String libraryUri = element != null ? element.getLibraryUri() : null;
+      String libraryUri = suggestion.getLibraryUri();
       String libraryUriToImport =
-        element != null && suggestion.getIsNotImported() != null && suggestion.getIsNotImported() ? element.getLibraryUri() : null;
+        element != null && suggestion.getIsNotImported() != null && suggestion.getIsNotImported() ? libraryUri : null;
```
Comment (scheglov) [inline Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java:233]: Would `suggestion.getIsNotImported() == true` or `suggestion.getIsNotImported() == Boolean.TRUE` work here?

Comment (alexander-doroshko) [reply Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java:233]: `suggestion.getIsNotImported() == true` will throw NPE, `suggestion.getIsNotImported() == Boolean.TRUE` is fine. I ended up with
`String libraryUriToImport = suggestion.getIsNotImported() == Boolean.TRUE ? libraryUri : null;`

File: Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AvailableSuggestion.java [hunk 2]
```diff
@@ -122,7 +122,7 @@ public boolean equals(Object obj) {
 
   public static AvailableSuggestion fromJson(JsonObject jsonObject) {
     String label = jsonObject.get("label").getAsString();
-    String declaringLibraryUri = jsonObject.get("declaringLibraryUri") != null ? jsonObject.get("declaringLibraryUri").getAsString() : null;
+    String declaringLibraryUri = jsonObject.get("declaringLibraryUri").getAsString();
```
Comment (alexander-doroshko) [inline Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AvailableSuggestion.java:125]: I'll keep this line unchanged just for the compatibility with older Dart SDK versions.

File: Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/Outline.java [hunk 3]
```diff
@@ -103,8 +103,8 @@ public static Outline fromJson(Outline parent, JsonObject outlineObject) {
     Element element = Element.fromJson(elementObject);
     int offset = outlineObject.get("offset").getAsInt();
     int length = outlineObject.get("length").getAsInt();
-    int codeOffset = outlineObject.get("codeOffset") == null ? offset : outlineObject.get("codeOffset").getAsInt();
-    int codeLength = outlineObject.get("codeLength") == null ? length : outlineObject.get("codeLength").getAsInt();
+    int codeOffset = outlineObject.get("codeOffset").getAsInt();
+    int codeLength = outlineObject.get("codeLength").getAsInt();
```
Comment (alexander-doroshko) [inline Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/Outline.java:107]: I'll keep these lines unchanged just for the compatibility with older Dart SDK versions.

**Comment by alexander-doroshko** [comment]
Pushed, thanks.

**Pull request closed without merging**